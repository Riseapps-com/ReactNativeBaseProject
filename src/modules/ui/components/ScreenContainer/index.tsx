import React from 'react';
import { StatusBar, StatusBarStyle, StyleProp, View, ViewProps, ViewStyle } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';

import { useTheme } from '~theme';

import themedStyles from './styles';

export type ScreenContainerProps = ViewProps & {
  statusBarStyle?: StatusBarStyle;
  withSafeTopMargin?: boolean;
  withSafeBottomMargin?: boolean;
  withSafeLeftMargin?: boolean;
  withSafeRightMargin?: boolean;
};

const ScreenContainer: React.FC<ScreenContainerProps> = props => {
  const [styles] = useTheme(themedStyles);
  const { top, bottom, left, right } = useSafeAreaInsets();
  const {
    statusBarStyle,
    withSafeTopMargin,
    withSafeBottomMargin,
    withSafeLeftMargin,
    withSafeRightMargin,
    ...restProps
  } = props;
  const containerStyles: StyleProp<ViewStyle> = [
    styles.container,
    props.withSafeTopMargin && { marginTop: top },
    props.withSafeBottomMargin && { marginBottom: bottom },
    props.withSafeLeftMargin && { marginStart: left },
    props.withSafeRightMargin && { marginEnd: right },
    props.style,
  ];

  return (
    <>
      <StatusBar barStyle={statusBarStyle || 'light-content'} />
      <View {...restProps} style={containerStyles}>
        {props.children}
      </View>
    </>
  );
};

export default ScreenContainer;
