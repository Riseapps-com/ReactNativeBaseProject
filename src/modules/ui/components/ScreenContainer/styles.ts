import { styleSheetFactory } from '~theme';

export default styleSheetFactory(theme => ({
  container: {
    flex: 1,
    backgroundColor: theme.background,
  },

  centered: {
    alignItems: 'center',
    justifyContent: 'center',
  },
}));
