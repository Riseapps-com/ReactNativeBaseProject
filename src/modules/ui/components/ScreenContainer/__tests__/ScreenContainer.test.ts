import { renderComponent } from '~modules/tests';

import ScreenContainer, { ScreenContainerProps } from '../index';

const renderScreenContainer = (props: ScreenContainerProps = {}) => renderComponent(ScreenContainer, props);

describe('ui', () => {
  describe('<ScreenContainer />', () => {
    it('renders <ScreenContainer />', () => {
      const screenContainer = renderScreenContainer();

      expect(screenContainer.getByA11yLabel('Screen container')).toBeTruthy();
      expect(screenContainer).toMatchSnapshot();
    });

    describe('safe area', () => {
      it('renders top margin if it is specified', () => {
        const screenContainer = renderScreenContainer({
          withSafeTopMargin: true,
        });

        expect(screenContainer.getByA11yLabel('Screen container')).toHaveStyle({
          marginTop: 8,
        });
      });

      it('renders bottom margin if it is specified', () => {
        const screenContainer = renderScreenContainer({
          withSafeBottomMargin: true,
        });

        expect(screenContainer.getByA11yLabel('Screen container')).toHaveStyle({
          marginBottom: 16,
        });
      });

      it('renders left margin if it is specified', () => {
        const screenContainer = renderScreenContainer({
          withSafeLeftMargin: true,
        });

        expect(screenContainer.getByA11yLabel('Screen container')).toHaveStyle({
          marginStart: 24,
        });
      });

      it('renders right margin if it is specified', () => {
        const screenContainer = renderScreenContainer({
          withSafeRightMargin: true,
        });

        expect(screenContainer.getByA11yLabel('Screen container')).toHaveStyle({
          marginEnd: 32,
        });
      });
    });
  });
});
