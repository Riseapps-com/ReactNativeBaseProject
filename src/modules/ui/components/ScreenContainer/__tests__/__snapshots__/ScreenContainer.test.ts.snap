// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui <ScreenContainer /> renders <ScreenContainer /> 1`] = `
<View
  accessibilityLabel="Screen container"
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
/>
`;
